specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json

```
@@ -1777,58 +1820,248 @@
+          "type": "string",
+          "enum": [
+            "Rank",
+            "DisplayName"
+          ],
+          "x-ms-enum": {
+            "name": "InstancesSortBy",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "SearchInstancesHierarchiesParameters": {
+      "description": "Parameter of how to return time series instance hierarchies by search instances call.",
+      "type": "object",
+      "properties": {
+        "expand": {
+          "description": "Definition of whether to expand hierarchy nodes in the same search instances call. Optional, default is 'UntilChildren'.",
+          "type": "object",
+          "$ref": "#/definitions/HierarchiesExpandParameter"
+        },
+        "sort": {
+          "description": "Definition of sorting of hierarchy nodes. Optional, default is 'CumulativeInstanceCount'.",
+          "type": "object",
+          "$ref": "#/definitions/HierarchiesSortParameter"
+        },
+        "pageSize": {
+          "description": "Maximum number of hierarchies in the same level to return in the same page. Optional, default is 10 when not set. Range is from 1 to 100. If there are results beyond the page size, the continuation token can be used to fetch the next page.",
+          "type": "integer",
+          "format": "int32"
+        }
+      }
+    },
+    "HierarchiesExpandParameter": {
+      "description": "Definition of whether to expand hierarchy nodes in the same search instances call.",
+      "type": "object",
+      "properties": {
+        "kind": {
+          "description": "Kind of the expansion of hierarchy nodes. When it is set to 'UntilChildren', the hierarchy nodes are expanded recursively until there is more than one child. When it is set to 'OneLevel', the hierarchies are expanded only at the single level matching path search instances parameter. Optional, default is 'UntilChildren'.",
+          "type": "string",
+          "enum": [
+            "UntilChildren",
+            "OneLevel"
+          ],
+          "x-ms-enum": {
+            "name": "HierarchiesExpandKind",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "HierarchiesSortParameter": {
+      "description": "Definition of sorting of hierarchy nodes.",
+      "type": "object",
+      "properties": {
+        "by": {
+          "description": "Value to use for hierarchy node sorting. When it is set to 'CumulativeInstanceCount', the returned hierarchies are sorted based on the total instances belonging to the hierachy node and its child hierarchy nodes. When it is set to 'Name', the returned hierarchies are sorted based on the hierarchy name. Optional, default is 'CumulativeInstanceCount'.",
```

hierachy -> hierarchy
```suggestion
          "description": "Value to use for hierarchy node sorting. When it is set to 'CumulativeInstanceCount', the returned hierarchies are sorted based on the total instances belonging to the hierarchy node and its child hierarchy nodes. When it is set to 'Name', the returned hierarchies are sorted based on the hierarchy name. Optional, default is 'CumulativeInstanceCount'.",
```